include/deal.II/matrix_free/fe_evaluation.h

```diff
@@ -1505,22 +1604,177 @@ class FEEvaluationAccess<1,1,Number> : public FEEvaluationBase<1,1,Number>
+ *         VectorizedArray<double> div = -trace(sym_grad_u);
+ *         pressure.submit_value   (div, q);
+ *
+ *         // subtract p * I
+ *         for (unsigned int d=0; d<dim; ++d)
+ *           sym_grad_u[d][d] -= pres;
+ *
+ *         velocity.submit_symmetric_gradient(sym_grad_u, q);
+ *      }
+ *
+ *     velocity.integrate (false,true);
+ *     velocity.distribute_local_to_global (dst.block(0));
+ *     pressure.integrate (true,false);
+ *     pressure.distribute_local_to_global (dst.block(1));
+ *   }
+ * @endcode
+ *
+ * This code assumes that a BlockVector of two components describes the
+ * velocity and pressure components, respectively. For identifying the
+ * different DoFHandler objects for velocity and pressure, the second argument
+ * to the FEEvaluation objects specify the respective component 0 for velocity
+ * and 1 for pressure. For further examples of vector-valued problems, the
+ * deal.II test suite includes a few additional examples as well, e.g. the
+ * Stokes operator described above is found at
+ * https://github.com/dealii/dealii/blob/master/tests/matrix_free/matrix_vector_stokes_noflux.cc
+ *
+ * <h3>Handling several integration tasks and data storage</h3>
+ *
+ * The design of FEEvaluation and MatrixFree separates the geometry from the
+ * basis functions. Therefore, several DoFHandler objects (or the same
+ * DoFHandler equipped with different constraint objects) can share the same
+ * geometry information like in the Stokes example above. All gemoetry is
+ * cached once in MatrixFree, so FEEvaluation does not need to do expensive
+ * initialization calls and rather sets a few pointers. This realization is
+ * based on the idea that the geometry information is needed only once also
+ * when several fields are evaluated, in a departure from FEValues which sets
+ * up the internal mapping data for each field. If for example a
+ * multi-component PDE involves the shape values on one component and the
+ * shape gradient on the other, no efficiency is lost if both are based on the
+ * same MatrixFree object where the update flags specify that both @p
+ * update_values , @p update_gradients , and @p update_jxw_values are
+ * given. The selection of desired quantities of shape values is through the
+ * flags in the evaluate() or integrate calls and the access at quadrature
+ * points:
+ *
+ * @code
+ * phi1.evaluate(true, false);
+ * phi2.evaluate(false, true);
+ * for (unsigned int q_index=0; q_index<phi1.n_q_points; ++q_index)
+ *   {
+ *     VectorizedArray<double> val1 = phi1.get_value(q);
+ *     Tensor<1,dim,VectorizedArray<double> > grad2 = phi2.get_gradient(q);
+ *     Point<dim,VectorizedArray<double> > point = phi1.get_quadrature_point(q);
+ *     // ... some complicated formula combining those three...
+ *   }
+ * @endcode
+ *
+ * In the loop over quadrature points, one can ask any of the two FEEvaluation
+ * objects mdash; it does not really matter which one because they only keep
```

Comment:
You are right, I did not put the correct html code in here. It should have been `&mdash;` 

I actually built the documentation for looking at this very part, but by the time I had built it I had forgotten about what I wanted to look for ... ;)